e2e-tests/src/test/java/uk/nhs/adaptors/gp2gp/e2e/EhrExtractTest.java

```diff
@@ -88,7 +91,30 @@ public void When_ExtractRequestReceivedForPatientWithNoDocs_Expect_DatabaseToBeU
         assertThatNotDocumentsWereAdded(gpcAccessDocument);
 
         var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get("ackToRequester"));
-        assertThatAcknowledgementToRequesterWasSent(ackToRequester);
+        assertThatAcknowledgementToRequesterWasSent(ackToRequester, ACCEPTED_ACKNOWLEDGEMENT_TYPE_CODE);
+    }
+
+    @Test
+    public void When_ExtractRequestReceivedForNotExistingPatient_Expect_ErrorUpdatedInDatabase() throws Exception {
+        String conversationId = UUID.randomUUID().toString();
+        String ehrExtractRequest = buildEhrExtractRequest(conversationId, NOT_EXISTING_PATIENT_NHS_NUMBER);
+
+        MessageQueue.sendToMhsInboundQueue(ehrExtractRequest);
+
+        var ehrExtractStatus = waitFor(() -> Mongo.findEhrExtractStatus(conversationId));
+        assertThatInitialRecordWasCreated(conversationId, ehrExtractStatus, NOT_EXISTING_PATIENT_NHS_NUMBER);
+
+        var ackToRequester = (Document) waitFor(() -> Mongo.findEhrExtractStatus(conversationId).get("ackToRequester"));
+        assertThatNegativeAcknowledgementToRequesterWasSent(ackToRequester, NEGATIVE_ACKNOWLEDGEMENT_TYPE_CODE);
+    }
+
+    private String buildEhrExtractRequest(String conversationId, CharSequence notExistingPatientNhsNumber) throws IOException {
```

Comment:
Out of curiosity, why is `notExistingPatientNhsNumber` of type `CharSequence`?

Reply:
inteliji did the refactor, I didn't notice.  :O